analyzere/base_resources.py

```diff
@@ -229,10 +229,17 @@ def upload_data(self, file_or_str):
         """
         file_obj = StringIO(file_or_str) if isinstance(
             file_or_str, six.string_types) else file_or_str
-        length = utils.file_length(file_obj)
 
-        # Initiate upload session
-        request_raw('post', self._data_path, headers={'Entity-Length': length})
+        # Upload file with known entity size if file object supports random
+        # access.
+        if hasattr(file_obj, 'seek'):
```

Comment:
Yes, on the `io` base class, but not for `StringIO` and although Python 3's `StringIO` implements `seekable()`, `six` does not seem to patch Python 2's version. So basically, `StringIO` on Python 2 and Python 3 with or without `six` have a different interface.